File: src/main/java/com/jloom/util/Tokens.java
package com.jloom.util;

import java.util.Map;

public final class Tokens {

    private Tokens() {
    }

    public static String substitute(String text, Map<String, String> tokens) {
        String result = text;
        for (Map.Entry<String, String> entry : tokens.entrySet()) {
            result = result.replace("{{" + entry.getKey() + "}}", entry.getValue());
        }
        return result;
    }
}

File: node/src/merge/tokens.ts
/** Mirrors util/Tokens.java: naive literal {{key}} -> value substring replacement, no escaping. */
export function substitute(text: string, tokens: Record<string, string>): string {
  let result = text;
  for (const [key, value] of Object.entries(tokens)) {
    result = result.split(`{{${key}}}`).join(value);
  }
  return result;
}
